### DISCUSSION TITLE
Include SHA256 hashes for easy integrity checking

### DISCUSSION THREAD

**Discussion by Chuckdor** [civitai/civitai#98] [Ideas]
I think it would be handy and useful to have in the information page the model SHA256 hash as an easy way to check integrity of model file. This would be handy for many reasons, first to make sure it is the original file and it hasn't been altered but also for integrity checking purposes after downloading through bad internet connections, as well as in case of faulty harddisks, to check against possible disk corruption.

**Comment by JustMaier** [comment]
We're working on this this week. We hope to have a SHA256 and the AUTO hash for each model. We'll probably need help from the community to get the hashes for models retroactively.

**Reply by Chuckdor** [reply]
Sorry for the late reply. That's awesome to read. And yeah, it's a lot of different models but nothing the community can't help with, that's for sure.

Thank you very much.

**Comment by JustMaier** [comment]
**This is live!**
![image](https://user-images.githubusercontent.com/607609/210707547-f49f133e-159d-4855-80b5-1391eda3e096.png)
![image](https://user-images.githubusercontent.com/607609/210707601-8c722e3f-6e7e-4483-91f8-eb7735ff9b85.png)

We display both hashes (just page through them by clicking the arrow), and you can click a hash to copy it. We still need to scan everything retroactively to get the old hashes, but hope to have that done some time next week.

As part of this we also added the ability to search by hash in both the API and the site.

**Reply by Chuckdor** [reply]
Oh, amazing. And really great idea to add them to the search engine too. Thank you very much. :)
